src/pytypes/onefuzztypes/models.py

```diff
@@ -841,6 +846,9 @@ class InstanceConfig(BaseModel):
 
     allowed_aad_tenants: List[UUID]
     network_config: NetworkConfig = Field(default_factory=NetworkConfig)
+    nsg_config: NetworkSecurityGroupConfig = Field(
```

Comment:
Should we name this field `proxy_nsg_config`?

Reply:
Personally I think we can keep it more general in case we have use for it in other contexts.

Reply:
Yeah, maybe! OTOH, that's a bit far out, and it's easy to just add another (more specific) field. Keeping it as-is for that reason kind of makes the bet that (1) we'll want another NSG, (2) that this unknown future NSG will want to have the _exact same settings_ as the proxy NSG. My guess is that is unlikely. I think there's also the risk of `network_config` (for the VMSS vnet) and `nsg_config` seeming related, when the NSG specifically _not_ for the VMSS network! So the extra specificity here might actually be a really helpful distinguisher.

To contrast with my reasoning elsewhere, where I thought it made more sense to keep the config class name as-is, and more general: if we _did_ add another NSG, which needed a config, "an NSG config" is a more generic concept. The fields of an an NSG config are determined by what Azure NSG resources provide, not their purpose in OneFuzz. In other words, the role of the NSG (which we provide) is unrelated to the configuration values (which we don't provide).

Reply:
I think that's a fair point. 

Do you want both to be changed? I.e. proxy_nsg_config & ProxyNetworkSecurityGroupConfig?

Reply:
Updated.

Reply:
👍 to what @bmc-msft said, that's exactly where my own thinking landed.